packages/engine-core/src/binary/BinaryEngine.ts

```diff
@@ -447,10 +447,24 @@ ${dim("In case we're mistaken, please report this to us 🙏.")}`)
       await this.stopPromise
     }
 
+    const retries = { times: 10 }
+    const retryInternalStart = async () => {
```

Comment:
```suggestion
    // Retries were added in https://github.com/prisma/prisma/pull/18874#issue-1679183725 to avoid these 2 flaky issues:
    // 1. Query engine exited with code 1
    // 2. Query engine exited with code 255
    // PANIC in /root/.cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.23/src/server/server.rs:79:13
    // error binding to 127.0.0.1:36437: error creating server listener: Address already in use (os error 98)
    //
    const retryInternalStart = async () => {
```